public/app/plugins/panel/geomap/editor/LayerEditor.tsx

```diff
@@ -85,6 +87,36 @@ export const LayerEditor: FC<LayerEditorProps> = ({ options, onChange, data, fil
           showIf: (opts: MapLayerOptions) => opts.location?.mode === FrameGeometrySourceMode.Geohash,
           // eslint-disable-next-line react/display-name
           // info: (props) => <div>HELLO</div>,
+        })
+        .addFieldNamePicker({
+          path: 'location.lookup',
+          name: 'Lookup Field',
+          settings: {
+            filter: (f: Field) => f.type === FieldType.string,
+            noFieldsMessage: 'No string fields found',
+          },
+          showIf: (opts: MapLayerOptions) => opts.location?.mode === FrameGeometrySourceMode.Lookup,
+        })
+        .addSelect({
+          path: 'location.lookupSrc',
+          name: 'Lookup Source Option',
+          settings: {
+            options: [
+              { value: LookupSourceOptions.Countries, label: 'Countries' },
```

Comment:
Super nitpick: can we change all  these to singular, just reads better that we lookup a Country using a country code rather than Countries, same for State via Sates etc ☀️

Reply:
It depends I read it wrong initially , read it as what does the field represent (country code or state etc) but now see this option is what data set to lookup the location in so maybe a plural makes sense in this case

Reply:
Yes, this option is for which json file to lookup the location.